src/components/AppProvider.tsx

```diff
@@ -19,23 +19,27 @@ interface Props {
   children: React.ReactNode
 }
 
-/**
- * Configuration for the Metabase provider.
- */
-const config: SDKConfig = {
-  metabaseInstanceUrl: METABASE_INSTANCE_URL,
-  jwtProviderUri: `${API_HOST}${JWT_PROVIDER_URI}`,
-  loaderComponent: MetabaseLoader,
-  errorComponent: MetabaseError,
-}
-
 export const AppProvider = ({ children }: Props) => {
   const [site] = useAtom(siteAtom)
 
   const theme = useMemo(() => {
     return SITE_CONFIG_MAP[site].metabase
   }, [site])
 
+  // Configuration for the Metabase provider.
+  const config: SDKConfig = useMemo(() => {
+    return {
+      metabaseInstanceUrl: METABASE_INSTANCE_URL,
+      jwtProviderUri: `${API_HOST}${JWT_PROVIDER_URI}`,
+      loaderComponent: MetabaseLoader,
+      errorComponent: MetabaseError,
+
+      // Append the current site as a query parameter to the JWT provider URL.
+      fetchRequestToken: (url: string) =>
+        fetch(`${url}?site=${site}`).then((res) => res.json()),
```

Comment:
We need the `site` query parameter in `/metabase/sso` to change depending on which site we are on, so we can wrap the SDK configuration in a `useMemo`.

Reply:
From the point of view of the manteiners of the sdk, i think it's safer/easier to keep `jwtProviderUri` constant and ask for all the dynamic logic in `fetchRequestToken`

Reply:
> From the point of view of the manteiners of the sdk, i think it's safer/easier to keep `jwtProviderUri` constant and ask for all the dynamic logic in `fetchRequestToken`

This isn't well communicated in our Readme though. If I'm the customer I'd expect an easy URL manipulation via `jwtProviderUri`.

Would it be hard to make that happen?

Reply:
It's not communicated at all and it probably even works, but I'm not sure we should tell people to do that, as I think it's easier for us to tell them that if they need anything dynamic they should use the function so that we only have to test that to work.
It's just for the sake of simplicity, if you feel it's more natural to have a dynamic url i'm okay with that but we'll have to write tests for both and make sure both ways work and will keep working in the future

Reply:
I think this could be a side-discussion. This doesn't have to be addressed here.

Reply:
@WiNloSt I completely forgot that we can pass a dynamic `jwtProviderUri`. Although, that won't work in the current SDK because of the `loginStatus.status` check in the current SDK - we only allow SDK auth config to be updated if the status is not yet initialized.

(It looks like that this check in the SDK is written incorrectly too - you aren't supposed to `return` early in `useEffect`, as returning is for sending the context)

```tsx
useEffect(() => {
  if (loginStatus.status !== "uninitialized") {
    return;
  }

  setupSdkAuth(config, dispatch);
}, [config, dispatch, loginStatus.status]);
```

Reply:
Let's keep this as is then. We should fix this separately.